giraph-core/src/main/java/org/apache/giraph/master/BspServiceMaster.java

```diff
@@ -1734,7 +1735,7 @@ private CheckpointStatus getCheckpointStatus(long superstep) {
     if (checkpointFrequency == 0) {
       return CheckpointStatus.NONE;
     }
-    long firstCheckpoint = INPUT_SUPERSTEP + 1 + checkpointFrequency;
+    long firstCheckpoint = INPUT_SUPERSTEP + 1;
```

Comment:
What is the reason for changing this? Do you want it to always do checkpoint after the first superstep?